docs/the-new-architecture/custom-cxx-types.md

```diff
@@ -0,0 +1,417 @@
+2. Add the following code to that file:
+
+```cpp title="Int64.h"
+#pragma once
+
+#include <react/bridging/Bridging.h>
+
+namespace facebook::react {
+
+template <>
+struct Bridging<int64_t> {
+  // Converts from the JS representation to the C++ representation
+  static int64_t fromJs(jsi::Runtime &rt, const jsi::String &value) {
+    try {
+      size_t pos;
+      auto str = value.utf8(rt);
+      auto num = std::stoll(str, &pos);
+      if (pos != str.size()) {
+        throw std::invalid_argument("Invalid number"); // don't support alphanumeric strings
+      }
+      return num;
+    } catch (const std::logic_error &e) {
+      throw jsi::JSError(rt, e.what());
+    }
+  }
+
+  // Converts from the C++ representation to the JS representation
+  static jsi::String toJs(jsi::Runtime &rt, int64_t value) {
+    return bridging::toJs(rt, std::to_string(value));
+  }
+};
+
+}
+```
+
+The key components for your custom bridging header are:
+
+- Explicit specialization of the `Bridging` struct for your custom type. In this case, the template specify the `int64_t` type.
+- A `fromJs` function to convert from the JS representation to the C++ representation
+- A `toJs` function to convert from the C++ representation to the JS representation
+
+:::note
+On iOS, remember to add the `Int64.h` file to the Xoced project.
+:::
+
+### 2. Modify the JS Spec
+
+Now, we can modify the JS spec to add a method that uses the new type. As usual, we can use either Flow or TypeScript for our specs.
+
+1. Open the `specs/NativeSampleTurbomodule`
+2. Modify the spec as it follows:
+
+<Tabs groupId="custom-int64" queryString defaultValue={constants.defaultJavaScriptSpecLanguages} values={constants.javaScriptSpecLanguages}>
+<TabItem value="typescript">
+
+```diff title="NativeSampleModule.ts"
+import {TurboModule, TurboModuleRegistry} from 'react-native';
+
+export interface Spec extends TurboModule {
+  readonly reverseString: (input: string) => string;
++  readonly cubicRoot: (input: string) => number;
+}
+
+export default TurboModuleRegistry.getEnforcing<Spec>(
+  'NativeSampleModule',
+);
+```
+
+</TabItem>
+<TabItem value="flow">
+
+```diff title="NativeSampleModule.js"
+// @flow
+import type {TurboModule} from 'react-native';
+import { TurboModuleRegistry } from "react-native";
+
+export interface Spec extends TurboModule {
+  +reverseString: (input: string) => string;
++  +cubicRoot: (input: string) => number;
+}
+
+export default (TurboModuleRegistry.getEnforcing<Spec>(
+  "NativeSampleModule"
+): Spec);
+```
+
+</TabItem>
+</Tabs>
+
+In this files, we are defining the function that needs to be implemented in C++.
+
+### 3. Implement the Native Code.
+
+Now, we need to implement the function that we declared in the JS specification.
+
+1. Open the `specs/NativeSampleModule.h` file and apply the following changes:
+
+```diff title="NativeSampleModule.h"
+#pragma once
+
+#include <AppSpecsJSI.h>
++ #include "Int64.h"
+#include <memory>
+#include <string>
```

Comment:
can we sort them and have all the diamon import before/after the quote includes?